### PR TITLE
feat(Search): Added Search input

### PR DESCRIPTION
**Pull request by nicmell**
### Search Input

Added search input, minor fixes on base input and others.

localhost_6006__path=_docs_components-search--docs (1)

### [IMPORTANT] PR Checklist

- [x] I am aware of standards and conventions adopted in this repository, defined in the [CONTRIBUTING.md file](https://github.com/mia-platform/design-system/blob/main/CONTRIBUTING.md)

#### PR conventions

Please make sure your PR complies with the following rules before submitting it.

- [x] PR title follows the `<type>(<scope>): <subject>` structure
- [x] The PR has been labeled according to the type of changes (e.g., enhancement, bug).

#### Additional code checks

Based on your changes, some of these checks may not apply. Please make sure to check the relevant items in the list.

- [x] Changes are covered by tests 
- [x] Changes to components are accessible and documented in the Storybook
- [x] Typings have been updated
- [x] New components are exported from the `index` file
- [x] New files include the Apache 2.0 License disclaimer
- [x] The browser console does not contain errors

### CODE REVIEW

File: src/components/Input/Input.tsx [hunk 1]
```diff
@@ -80,5 +80,5 @@ export const Input = (
   )
 }
 
-Input.Type = BaseInput.Appearance
-Input.HTMLType = Type
+Input.Appearance = BaseInput.Appearance
+Input.Type = Type
```
[the remark this answers is not in the document]
Comment (nicmell) [reply src/components/Input/Input.tsx:84]: yep, it was an oversight. We decided to rename the property in a previous pr and forgot to update those assignments. The version containing the error is not used by website, so not a big deal.

File: src/components/Search/Search.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        isDisabled={isDisabled}
+        isError={isError}
+        isFullWidth={isFullWidth}
+        isReadOnly={isReadOnly}
+      />
+    </BaseInput>
+  )
+}
+
+Search.Type = Appearance
```
[the remark this answers is not in the document]
Comment (nicmell) [reply src/components/Search/Search.tsx:97]: Screenshot 2024-10-30 alle 11 13 21

File: src/components/InputNumber/InputNumber.tsx [hunk 3]
```diff
@@ -73,4 +73,4 @@ export const InputNumber = (
   )
 }
 
-InputNumber.Type = Appearance
+InputNumber.Appearance = Appearance
```
[the remark this answers is not in the document]
Comment (nicmell) [reply src/components/InputNumber/InputNumber.tsx:76]: Nope, they all have been fixed.

[the remark this answers is not in the document]
Comment (Danielecina) [reply src/components/InputNumber/InputNumber.tsx:76]: Screenshot 2024-10-30 alle 11 22 21

File: src/components/Search/Search.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Danielecina) [reply src/components/Search/Search.tsx]: ok, this does not answer the question of the correct export

**Review by Danielecina — DISMISSED** [review]

File: src/components/InputNumber/InputNumber.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicmell) [reply src/components/InputNumber/InputNumber.tsx:76]: homer-simpson-les-simpson

Fixed

File: src/components/Search/Search.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nicmell) [reply src/components/Search/Search.tsx]: ok got it. fixed

**Pull request merged**